core/codenvy-hosted-sso-server-organization/src/main/java/com/codenvy/auth/sso/server/OrgServiceUserValidator.java

```diff
@@ -36,14 +40,17 @@
     private final UserDao           userDao;
     private final UserNameValidator userNameValidator;
     private final boolean           userSelfCreationAllowed;
+    private final Set<String>       reservedNames;
 
     @Inject
     public OrgServiceUserValidator(UserDao userDao,
                                    UserNameValidator userNameValidator,
-                                   @Named("user.self.creation.allowed") boolean userSelfCreationAllowed) {
+                                   @Named("user.self.creation.allowed") boolean userSelfCreationAllowed,
+                                   @Named("user.reserved_names") String[] reservedNames) {
         this.userDao = userDao;
         this.userNameValidator = userNameValidator;
         this.userSelfCreationAllowed = userSelfCreationAllowed;
+        this.reservedNames = Sets.newHashSet(Arrays.asList(reservedNames));
```

Comment:
with guava you could give the array directly
http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Sets.html#newHashSet(E...)